File: grpcio-py/benchmark.py
#!/usr/bin/env python

from __future__ import print_function

import sys

from datetime import datetime, timedelta
from client import create_client, single_daq_run, shutdown_client


def format_delta(delta_time):
    return "%6.3fs" % (float(delta_time.seconds) +
                       (float(delta_time.microseconds) / 1E6))


def main():
    host = 'localhost'
    port = 50051
    reps = 1000

    if len(sys.argv) > 1:
        reps = int(sys.args[1])

    init_start = datetime.now()

    client = create_client(host, port)

    init_time = datetime.now() - init_start

    print("Run %d reps" % reps)

    run_start = datetime.now()

    for idx in range(reps):
        single_daq_run(client, False)

    run_time = datetime.now() - run_start

    stop_start = datetime.now()

    shutdown_client(client)

    stop_time = datetime.now() - stop_start

    print("Init: %s      Run: %s      Stop: %s      Total: %s" %
          (format_delta(init_time), format_delta(run_time),
           format_delta(stop_time),
           format_delta(init_time + run_time + stop_time)))


if __name__ == "__main__":
    main()

File: grpc-java/src/main/java/org/glowacki/grpc/Benchmark.java
package org.glowacki.grpc;

public class Benchmark
{
    private static final int port = 50051;

    private static final String formatNano(long nanoseconds)
    {
        final double seconds = ((double) nanoseconds) / 1E9;
        return String.format("%6.3fs", seconds);
    }

    public static void main(String[] args)
        throws Exception
    {
        int reps = 1000;
        if (args.length >= 1) {
            reps = Integer.parseInt(args[0]);
        }

        final long initStart = System.nanoTime();

        Client client = new Client("localhost", port);

        final long initTime = System.nanoTime() - initStart;

        System.out.println("Run " + reps + " reps");

        final long runStart = System.nanoTime();

        for (int i = 0; i < reps; i++) {
            client.singleDAQRun(false);
        }

        final long runTime = System.nanoTime() - runStart;

        final long stopStart = System.nanoTime();
        client.shutdown();
        final long stopTime = System.nanoTime() - stopStart;

        System.out.print("Init:  " + formatNano(initTime));
        System.out.print("      Run:   " + formatNano(runTime));
        System.out.print("      Stop:  " + formatNano(stopTime));
        System.out.println("      Total: " +
                           formatNano(initTime + runTime + stopTime));
    }
}
